src/callBuilder.js

```diff
@@ -0,0 +1,139 @@
+import {NotFoundError, NetworkError, BadRequestError} from "./errors";
+
+let URI = require("URIjs");
+let URITemplate = require('URIjs/src/URITemplate');
+
+
+let axios = require("axios");
+var EventSource = (typeof window === 'undefined') ? require('eventsource') : window.EventSource;
+let toBluebird = require("bluebird").resolve;
+
+/**
+* @class Builder
+*/
+export class CallBuilder {
+
+    /*
+    * @constructor
+    */
+    constructor(url) {
+        this.url = url;
+        this.filter = [];
+    }
+
+    checkFilter() {
+        if (this.filter.length >= 2) {
+            throw new BadRequestError("Too many filters specified", this.filter);
+        } 
+        if (this.filter.length === 1) {
+            this.url.segment(this.filter[0]);
+        }        
+    }
+
+    call() {
+        this.checkFilter();
+        var promise = this._sendNormalRequest(this.url)
+            .then(this._parseResponse.bind(this));
+        return promise;
+    }
+
+    /**
+    * Convert each link into a function on the response object.
+    */
+    _parseRecord(json) {
+        if (!json._links) {
+            return json;
+        }
+        var self = this;
+        var linkFn = function (link) {
+            return function (opts) {
+                if (link.template) {
+                    let template = URITemplate(link.href);
+                    return self._sendNormalRequest(URI(template.expand(opts)));
+                } else {
+                    return self._sendNormalRequest(URI(link.href));
+                }
+            };
+        };
+        Object.keys(json._links).map(function(value, index) {
+            var link = json._links[value];
+            json[value] = linkFn(link);
+        });
+        return json;
+    }
+    
+    _sendNormalRequest(url) {
+        // To fix:  #15 Connection Stalled when making multiple requests to the same resource
```

Comment:
FYI:  using a `TODO` label is convention for comments like this.   Following the convention makes it easy for someone looking to fix something random... just grep for TODO